odoo/addons/test_new_api/models/test_new_api.py

```diff
@@ -182,6 +183,14 @@ def _compute_size(self):
         for message in self:
             message.size = len(message.body or '')
 
+    @api.depends('attributes')
+    def _compute_is_attributes_empty(self):
```

Comment:
The computed field depends on the property field itself, and not its definition.  The dependency on the definition is an artefact of the property field's implementation.  You have to fix how the property field deals with assignments on _new_ records, because in that context, you may _not_ modify the property's definition field.

Reply:
see other comment